gradio/interface.py

```diff
@@ -487,8 +486,11 @@ def clean_html(raw_html):
             if component.label is None:
                 component.label = "output_" + str(i)
 
-        self.cache_examples = cache_examples
-        if cache_examples:
+        if self.is_space and cache_examples is None:
+            self.cache_examples = True
+        else:
+            self.cache_examples = cache_examples
```

Comment:
Same as above, consider

```suggestion
            self.cache_examples = cache_examples or False
```